src/onepiece_classify/models/build_model.py

```diff
@@ -3,20 +3,50 @@
 from torchvision import models
 
 
-def create_model(
-    num_classes: int,
-    seed: int=42
-):
-    model = models.mobilenet_v3_large(weights="DEFAULT")
-
-    for param in model.parameters():
-        param.required_grad = False
-
-    # change model head
-    torch.manual_seed(seed)
-    model.classifier = nn.Sequential(
-        nn.Dropout(p=0.2),
-        nn.Linear(in_features=960, out_features=num_classes)
-    )
-
-    return model
\ No newline at end of file
+# def create_model(
+#     num_classes: int,
+#     seed: int=42
+# ):
+#     model = models.mobilenet_v3_large(weights="DEFAULT")
+
+#     for param in model.parameters():
+#         param.required_grad = False
+
+#     # change model head
+#     torch.manual_seed(seed)
+#     model.classifier = nn.Sequential(
+#         nn.Dropout(p=0.2),
+#         nn.Linear(in_features=960, out_features=num_classes)
+#     )
+
+#     return model
+
+class ImageRecogModel(nn.Module):
+    
+    def __init__(self, in_features, num_classes):
+        self.in_features = in_features
+        self.num_classes = num_classes
+
+        self.backbone = _build_backbone()
+        self.dropout = nn.Dropout(p=0.2)
+
+    def _build_backbone(self) -> models:
+        model = models.mobilenet_v3_large(weights="DEFAULT")
+
+        for param in model.parameters():
+            param.requires_grad = False
+
+        return model
+
+    def forward(self) -> torch.Tensor:
+        # x = self.backbone()
+        # x = self.dropout(nn.Linear(self.in_features, self.num_classes))
+        self.backbone.classifier = nn.Sequential(
```

Comment:
what the heck is this!!!

why you place an object inside a forward, why you initialize object here???

this method only for execution of the object that has been initialized in the constructor of the model